src/alley/wp/simplecache/class-metadata-adapter.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * Metadata_Adapter class file
+ *
+ * @package wp-psr16
+ */
+
+namespace Alley\WP\SimpleCache;
+
+use Psr\SimpleCache\CacheInterface;
+use Symfony\Component\Clock\NativeClock;
+
+/**
+ * PSR-16 implementation that caches data in WordPress object metadata.
+ */
+final class Metadata_Adapter implements CacheInterface {
+	/**
+	 * Constructor.
+	 *
+	 * @param string $type Object type.
+	 * @param int    $id   Object ID.
+	 */
+	private function __construct(
+		private readonly string $type,
+		private readonly int $id,
+	) {}
+
+	/**
+	 * Create an instance using the default composition.
+	 *
+	 * @param string $type Object type.
+	 * @param int    $id   Object ID.
+	 * @return CacheInterface
+	 */
+	public static function create( string $type, int $id ): CacheInterface {
+		return new PSR16_Compliant(
+			new NativeClock(),
+			new Prefixed_Keys(
+				'_psr16_',
+				new Maximum_Key_Length(
+					255,
+					new self(
+						$type,
+						$id,
+					),
+				),
+			),
+		);
+	}
+
+	/**
+	 * Create an instance for caching to post meta.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return CacheInterface
+	 */
+	public static function for_post( int $post_id ): CacheInterface {
+		return self::create( 'post', $post_id );
+	}
+
+	/**
+	 * Create an instance for caching to term meta.
+	 *
+	 * @param int $term_id Term ID.
+	 * @return CacheInterface
+	 */
+	public static function for_term( int $term_id ): CacheInterface {
+		return self::create( 'term', $term_id );
+	}
+
+	/**
+	 * Fetches a value from the cache.
+	 *
+	 * @throws \Psr\SimpleCache\InvalidArgumentException If the $key string is not a legal value.
+	 *
+	 * @param string $key     The unique key of this item in the cache.
+	 * @param mixed  $default Default value to return if the key does not exist.
+	 * @return mixed The value of the item from the cache, or $default in case of cache miss.
+	 */
+	public function get( string $key, mixed $default = null ): mixed {
+		$get = $this->getMultiple( [ $key ], $default );
+
+		$out = $default;
+
+		foreach ( $get as $value ) {
+			$out = $value;
```

Comment:
Is it possible that getMultiple would ever return more than one value for this key?

Reply:
Possible, I guess, but only in the case of an implementation error against the spec. I'll update it just in case.